### PR TITLE
feat(insured-bridge): Block instant relays post-expiry and change relayDeposit interface to match relayAndSpeedUp

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

`relayAndSpeedUp` takes a `DepositData` struct as first param, while `relayDeposit` takes in de-structured `DepositData` params. The latter can easily take in a `DepositData` struct.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/contracts/insured-bridge/BridgePool.sol [hunk 1]
```diff
@@ -355,45 +353,21 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, Lockable {
...
-        uint64 depositId,
-        address payable l1Recipient,
-        address l2Sender,
-        uint256 amount,
-        uint64 slowRelayFeePct,
-        uint64 instantRelayFeePct,
-        uint32 quoteTimestamp,
-        uint64 realizedLpFeePct
-    ) public nonReentrant() {
+    function relayDeposit(DepositData memory depositData, uint64 realizedLpFeePct) public nonReentrant() {
```
Comment (nicholaspai) [inline packages/core/contracts/insured-bridge/BridgePool.sol:361]: Function params can be grouped conveniently into `depositData`

File: packages/core/contracts/insured-bridge/BridgePool.sol [hunk 2]
```diff
@@ -457,11 +433,12 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, Lockable {
     function speedUpRelay(DepositData memory depositData, RelayData memory relayData) public nonReentrant() {
         bytes32 depositHash = _getDepositHash(depositData);
         _validateRelayDataHash(depositHash, relayData);
-
         bytes32 instantRelayHash = _getInstantRelayHash(depositHash, relayData);
         require(
             // Can only speed up a pending relay without an existing instant relay associated with it.
-            relayData.relayState == RelayState.Pending && instantRelays[instantRelayHash] == address(0),
+            getCurrentTime() < relayData.priceRequestTime + relayData.liveness &&
```
Comment (nicholaspai) [inline packages/core/contracts/insured-bridge/BridgePool.sol:439]: New require condition: cannot speed up relays post liveness

File: packages/core/contracts/insured-bridge/BridgePool.sol [hunk 3]
```diff
@@ -112,6 +116,7 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, Lockable {
         uint32 priceRequestTime;
         uint256 proposerBond;
         uint256 finalFee;
+        uint32 liveness;
```
Comment (mrice32) [inline packages/core/contracts/insured-bridge/BridgePool.sol:119]: @nicholaspai I think most of the changes in this are good/worthwhile (blocking speed up after liveness, comment clarifications), but I think this liveness one adds unnecessary complexity to the implementation for something that doesn't seem to break anything or have any major downsides (by my estimation). Thoughts on dropping that part?

File: packages/core/contracts/insured-bridge/BridgePool.sol [hunk 4]
```diff
@@ -355,45 +357,21 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, Lockable {
...
-        uint64 slowRelayFeePct,
-        uint64 instantRelayFeePct,
-        uint32 quoteTimestamp,
-        uint64 realizedLpFeePct
-    ) public nonReentrant() {
+    function relayDeposit(DepositData memory depositData, uint64 realizedLpFeePct) public nonReentrant() {
         // The realizedLPFeePct should never be greater than 0.5e18 and the slow and instant relay fees should never be
         // more than 0.25e18 each. Therefore, the sum of all fee types can never exceed 1e18 (or 100%).
-        require(slowRelayFeePct < 0.25e18 && instantRelayFeePct < 0.25e18 && realizedLpFeePct < 0.5e18);
+        require(
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply packages/core/contracts/insured-bridge/BridgePool.sol:368]: linter

File: packages/core/contracts/insured-bridge/BridgePool.sol (hunk 3 above)
Comment (nicholaspai) [reply packages/core/contracts/insured-bridge/BridgePool.sol:119]: yeah it should

**Review by chrismaree — APPROVED** [review]
nice! I like the syntax change. also, good catch on the timer issues

**Pull request merged**